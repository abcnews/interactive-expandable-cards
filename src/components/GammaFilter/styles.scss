.filter {
  position: absolute;
  opacity: 0;
  width: 1px;
  height: 1px;
}
