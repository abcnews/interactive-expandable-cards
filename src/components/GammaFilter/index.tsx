import { h, FunctionalComponent } from 'preact';
import styles from './styles.scss';

type GammaFilterProps = {
  id: string;
  exponents: { r: number; g: number; b: number };
};

const GammaFilter: FunctionalComponent<GammaFilterProps> = ({ id, exponents: { r, g, b } }) => {
  return (
    <svg className={styles.filter}>
      <defs></defs>
      <filter id={id} color-interpolation-filters="sRGB" x="0" y="0" height="100%" width="100%">
        <feComponentTransfer>
          {/*
                       // @ts-ignore */}
          <feFuncR type="gamma" exponent={r} />
          {/*
                       // @ts-ignore */}
          <feFuncG type="gamma" exponent={g} />
          {/*
                       // @ts-ignore */}
          <feFuncB type="gamma" exponent={b} />
        </feComponentTransfer>
      </filter>
    </svg>
  );
};

export default GammaFilter;
