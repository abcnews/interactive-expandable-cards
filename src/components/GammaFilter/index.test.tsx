import { h } from 'preact';
import render from 'preact-render-to-string';

import GammaFilter from '.';

describe('GammaFilter', () => {
  test('It renders', () => {
    const actual = render(<GammaFilter id="id" exponents={{ r: 10, g: 10, b: 10 }} />);
    expect(actual).toMatchSnapshot();
  });
});
