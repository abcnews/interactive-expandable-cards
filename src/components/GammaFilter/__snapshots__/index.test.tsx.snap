// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GammaFilter It renders 1`] = `"<svg><defs></defs><filter id=\\"id\\" color-interpolation-filters=\\"sRGB\\" x=\\"0\\" y=\\"0\\" height=\\"100%\\" width=\\"100%\\"><feComponentTransfer><feFuncR type=\\"gamma\\" exponent=\\"10\\"></feFuncR><feFuncG type=\\"gamma\\" exponent=\\"10\\"></feFuncG><feFuncB type=\\"gamma\\" exponent=\\"10\\"></feFuncB></feComponentTransfer></filter></svg>"`;
